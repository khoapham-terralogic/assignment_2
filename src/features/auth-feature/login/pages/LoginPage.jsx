import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux'
import { loginUser } from '../../../../redux/actions/authAction';
import LoginPageInner from './LoginPageInner';

const LoginPage = ({ isLoading, loginUser, isAuth }) => {
    return (
        <LoginPageInner isAuth={isAuth} isLoading={isLoading} loginUser={loginUser} />
    );
}

LoginPage.propTypes = {
    loginUser: PropTypes.func.isRequired,
    isLoading: PropTypes.bool.isRequired,
    isAuth: PropTypes.bool.isRequired
}

export default connect(
    state => ({
        isAuth: state.auth.isAuth,
        isLoading: state.auth.isLoading
    }),
    { loginUser }
)(LoginPage);
