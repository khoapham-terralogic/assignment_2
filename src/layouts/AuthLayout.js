import React from 'react';
import { Switch, Route, Redirect, useRouteMatch } from 'react-router-dom'
import LoginPage from '../pages/Login/LoginPageConnectWrapper'
import RegisterPage from '../pages/Register/RegisterPageConnectWrapper';

import { Logo } from '../components'
import { rightImage } from '../constants'

const AuthLayout = () => {
    const match = useRouteMatch();
    return (
        <section className="auth">
            <div className="auth__container">
                <div className="auth__container__left">
                    <Logo />
                    <Switch>
                        <Redirect from="/auth" exact to="/auth/login" />
                        <Route path={`${match.url}/login`} exact render={() => <LoginPage />} />
                        <Route path={`${match.url}/register`} exact render={() => <RegisterPage />} />
                    </Switch>
                </div>
                <div className="auth__container__right">
                    <img className="auth__container__right-img" src={rightImage} alt="img" />
                </div>
            </div>
        </section>
    );
}

export default AuthLayout;
