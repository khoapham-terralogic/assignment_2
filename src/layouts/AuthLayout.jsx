import React from 'react';
import { Switch, Route, Redirect, useRouteMatch } from 'react-router-dom'
import LoginPage from '../features/auth-feature/login/pages/LoginPage'
import RegisterPage from '../features/auth-feature/register/pages/RegisterPage';

import { Logo } from '../components'
import { rightImage } from '../constants'

const AuthLayout = () => {
    const match = useRouteMatch();
    return (
        <section className="auth">
            <div className="auth__container">
                <div className="auth__container__left">
                    <Logo />
                    <Switch>
                        <Redirect from="/auth" exact to="/auth/login" />
                        <Route path={`${match.url}/login`} exact render={() => <LoginPage />} />
                        <Route path={`${match.url}/register`} exact render={() => <RegisterPage />} />
                    </Switch>
                </div>
                <div className="auth__container__right">
                    <img className="auth__container__right-img" src={rightImage} alt="img" />
                </div>
            </div>
        </section>
    );
}

export default AuthLayout;
