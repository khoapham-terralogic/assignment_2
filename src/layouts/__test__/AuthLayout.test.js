import React from 'react';
import { mount } from 'enzyme';
import AuthLayout from '../AuthLayout';
import { Router, MemoryRouter } from 'react-router-dom'
import { createBrowserHistory } from 'history'
import LoginPage from '../../pages/Login/LoginPageConnectWrapper';
const history = createBrowserHistory("/auth/login")
describe('Test <AuthLayout />', () => {
    it('should render', () => {
        const wrapper = mount(
            <Router history={history}>
                <AuthLayout />
            </Router>
        )
        expect(wrapper.find('[data-testid="logo-title"]').text()).toEqual("start your personal photo experience")
        expect(wrapper).toMatchSnapshot()
    });
    it('should render Login if path is /auth/login', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/auth/login']}>
                <AuthLayout />
            </MemoryRouter>
        );
        expect(wrapper.find(LoginPage)).toHaveLength(1)
    })
})
