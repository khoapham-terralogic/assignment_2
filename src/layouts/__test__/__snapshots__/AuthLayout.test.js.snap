// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <AuthLayout /> should render 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <AuthLayout>
    <section
      className="auth"
    >
      <div
        className="auth__container"
      >
        <div
          className="auth__container__left"
        >
          <Logo>
            <div
              className="logo"
              data-testid="logo"
            >
              <NavLink
                to="/"
              >
                <Link
                  aria-current="page"
                  className="active"
                  style={Object {}}
                  to={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": null,
                    }
                  }
                >
                  <LinkAnchor
                    aria-current="page"
                    className="active"
                    href="/"
                    navigate={[Function]}
                    style={Object {}}
                  >
                    <a
                      aria-current="page"
                      className="active"
                      href="/"
                      onClick={[Function]}
                      style={Object {}}
                    >
                      <img
                        alt="logo"
                        className="logo-img"
                        src="brand-logo.svg"
                      />
                    </a>
                  </LinkAnchor>
                </Link>
              </NavLink>
              <p
                className="logo-title"
                data-testid="logo-title"
              >
                start your personal photo experience
              </p>
            </div>
          </Logo>
          <Switch />
        </div>
        <div
          className="auth__container__right"
        >
          <img
            alt="img"
            className="auth__container__right-img"
            src="solution-experts.png"
          />
        </div>
      </div>
    </section>
  </AuthLayout>
</Router>
`;
